src/encoded/loadxl.py

```diff
@@ -81,17 +81,38 @@ def close(self):
             logger.error('load_data: failed to load with iter_response', error=self.caught)
 
 
-def load_data_via_ingester(vapp, ontology_json):
+def load_data_via_ingester(vapp: VirtualApp, ontology: dict, itype: str = "ontology_term") -> dict:
     """
     Entry point for call from encoded.ingester.processors.handle_ontology_update (2023-03-08).
+    Returns dictionary itemizing the created (post), updated (patch), skipped (skip), and
+    errored (error) ontology term uuids; as well as a count of the number of unique uuids processed.
     """
-    return Response(
-        content_type="text/plain",
-        app_iter=LoadGenWrapper(
-            load_all_gen(vapp, ontology_json, None, overwrite=True,
-                         itype="ontology_term", from_json=True, patch_only=False)
-        )
-    )
+    response = load_all_gen(vapp, ontology, None, overwrite=True, itype=itype, from_json=True, patch_only=False)
+    results = {"post": [], "patch": [], "skip": [], "error": []}
+    unique_uuids = set()
+    for item in response:
+        # ASSUME each item in the response looks something like one of (string or bytes):
+        # POST: 15425d13-01ce-4e61-be5d-cd04401dff29
+        # PATCH: 5b45e66f-7b4f-4923-824b-d0864a689bb
+        # SKIP: 4efe24b5-eb17-4406-adb8-060ea2ae2180
+        # ERROR: 906c4667-483e-4a08-96b9-3ce85ce8bf8c
+        if isinstance(item, bytes):
+            item = item.decode("ascii")
+        if not isinstance(item, str):
+            continue
+        item_split = item.split()
+        if len(item_split) != 2:
+            continue
+        action = item_split[0].lower()
+        if action.endswith(":"):
+            action = action[:-1]
+        uuid = item_split[1]
```

Comment:
Should we do a `log.error(...)` before either of these `continue` statements? Why would these happen?

In fact, though, given all of the meticulous picking apart, it seems like we should be using (untested):
```
INGESTION_RESPONSE_PATTERN = "^([A-Z]+): ([0-9a-f-]+)$"
```
somewhere earlier and then
```
        m = INGESTION_RESPONSE_PATTERN.match(item, str)
        if not m:
            log.error(f"load_data_via_ingester: response item in unexpected form: {item!r}")
            continue
        action, uuid = m.groups()
```
 Or we could treat the result as some kind of `"error"` in the result dictionary....?

@aschroed or @willronchetti  have a thought on this, too.

Reply:
I have also been curious as to what is included in the response - the reporting from loadxl has always been a bit opaque and this type of formatting of the response was something that Koray had done to try to get needed info about what happened.  I haven't looked at any of this recently but I assume that the continues were added because there was additional info in the response that did not give relevant info regarding what happened to a particular item.